src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp

```diff
@@ -260,34 +306,84 @@ struct fully_connected_onednn : typed_primitive_onednn_impl<fully_connected> {
-                attr->set_scales(DNNL_ARG_WEIGHTS, 1 << 1, dnnl::memory::dims{}, data_type);
+                ds_data_type = convert_data_type(arg.get_dependency(decompression_scale_idx).get_output_layout().data_type);
+                if (!is_four_bit_weight) {
+                    // 8-bit quantized weight
+                    attr->set_scales(DNNL_ARG_WEIGHTS, 1 << 1, dnnl::memory::dims{}, ds_data_type);
+                } else {
+                    // OneDNN does not support scalar zero-point for s4 and u8 type. Need to broadcast it.
+                    auto ifm = arg.get_dependency(1).get_output_layout().get_dim(1);
+                    auto ngroups = arg.get_dependency(decompression_scale_idx).get_output_layout().get_dim(1);
+                    group_size = ifm / ngroups;
+                    attr->set_scales(DNNL_ARG_WEIGHTS, (1 << 1) + (1 << 0), {group_size, 1}, ds_data_type);
+                }
             }
 
             if (prim->decompression_zero_point_scalar.has_value()) {
-                OPENVINO_ASSERT(!prim->decompression_zero_point_scalar.has_value(), "[GPU] OneDNN can't use scalar as a zero point value\n");
+                OPENVINO_ASSERT(!prim->decompression_zero_point_scalar.has_value(), "[GPU] UNIMPLEMENTED: scalar zp support\n");
             } else if (!prim->decompression_zero_point.empty()) {
                 auto decompression_zp_idx = !arg.bias_term() ? 3 : 4;
-                auto data_type = convert_data_type(arg.get_dependency(decompression_zp_idx).get_output_layout().data_type);
-                attr->set_zero_points(DNNL_ARG_WEIGHTS, 1 << 1, dnnl::memory::dims{}, data_type);
+                auto &zp_node = arg.get_dependency(decompression_zp_idx).as<data>();
+                memory::ptr zp_old_mem = zp_node.get_attached_memory_ptr();
+                zp_mem = zp_old_mem;
+
+                if (!is_four_bit_weight) {
+                    // 8-bit quantized weight
+                    auto decompression_zp_idx = !arg.bias_term() ? 3 : 4;
+                    dzp_data_type = convert_data_type(arg.get_dependency(decompression_zp_idx).get_output_layout().data_type);
+                    attr->set_zero_points(DNNL_ARG_WEIGHTS, 1 << 1, dnnl::memory::dims{}, dzp_data_type);
+                } else {
+                    // OneDNN does not support scalar zero-point for s4 and u8 type. Need to broadcast it.
+                    auto decompression_scale_idx = !arg.bias_term() ? 2 : 3;  // it assumes we have decompress_scale
+                    auto &scale_node = arg.get_dependency(decompression_scale_idx).as<data>();
+                    auto broadcasted_layout = scale_node.get_output_layout();
+                    broadcasted_layout.data_type = data_types::u8;
+                    dzp_data_type = convert_data_type(broadcasted_layout.data_type);
+
+                    if (zp_node.get_output_layout().get_linear_size() == 1) {
```

Comment:
Could you please explain why do we have ZP dependency for `get_linear_size() == 1` case? 
It seems compressed FC primitive will not have zp dependency in case of scalar ZP value: `has_scalar_zp ? "" : zp_name`:
```
auto fc = cldnn::fully_connected(primitive_name,
                                     cldnn::input_info(input_name),
                                     weights_name,
                                     bias_name,
                                     scale_name,
                                     has_scalar_zp ? "" : zp_name,
                                     cldnn::element_type_to_data_type(op->get_output_element_type(0)),
                                     cldnn::padding(),
                                     op->get_input_partial_shape(0).size(),
                                     op->get_input_partial_shape(1).size());
```
Am I missing something?

Reply:
decompression_zero_point_scalar is handled separately as "unimplemented" above. I did not handle it because 1) I did not have any usecase for it in LLM scenario and 2) it was unclear how we can handle that(onednn does not support f16 zp).

Reply:
Hmm, doesn't `zp_node.get_output_layout().get_linear_size() == 1` mean that we have single scalar zp value?..

I'm not sure what is the default mode of NNCF INT4 weights compression, but at least for INT4_SYM [it widely uses scalar zero point](https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/CompressWeights.md): `The primary precision in case of INT4_SYM mode is unsigned 4-bit integer and weights are quantized to it symmetrically with a fixed zero point equals to 8.`

Do you have some change which forces to treat this single value as not a scalar value stored in `decompression_zero_point_scalar`, but as a single value in the memory buffer referenced as `zp_node` FC's input?

Reply:
I did not add any logic for that. But when I checked our target models of chatglm or qwen, they created single-element zp-node. I'm not sure about the exact behavior of that part too..

Reply:
Oh, I get it. We have a bug, which prevents to recognize and save this single value to `decompression_zero_point_scalar`. After introducing FC with bias internal op, we have one more input at compressed FC primitive creating function:
```
FC input 0: RMS __module.model.layers.0.input_layernorm/aten::mul/Multiply_1_compressed_to_f16
FC input 1: Constant Constant_250023
FC input 2: Placeholder Placeholder_246417   // <- this is placeholder for bias input
FC input 3: Constant Constant_250021
FC input 4: Constant Constant_250022
```
But the logic which checks for a scalar value was not updated accordingly, and expects exactly 4 inputs:
```
    if (op->get_input_size() == 4) {
        auto zp_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(op->get_input_node_shared_ptr(3));
   ...
```

It looks like you can update your broadcast logic to allow both memory buffer with single value and `decompression_zero_point_scalar`, and then we may fix the above issue

Reply:
Applied, thanks!